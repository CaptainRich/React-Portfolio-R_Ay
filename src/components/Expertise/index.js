import React, {useState} from 'react'
import ReactCardFlip from "react-card-flip";
import inspectImage from "../../assets/images/mag-glass.svg";

import {HiOutlineDesktopComputer} from "react-icons/hi";
import {FcEngineering} from "react-icons/fc";



function Expertise() {

    const [isFlipped, setIsFlipped] = useState(false);

    const handleFlip = () => {
        // This (onClick) handler will toggle the state.
        setIsFlipped( !isFlipped );
    };

  return (
    <section className="my-5" >
      <h1 id="expertise">Capabilities & Skills ...</h1>
      <h1><p></p> </h1>

      <div className="container">

        {/* Define the link to present my resume. */}
        <div className="my-resume">
          {/* Configure this link to open in a new "tab" */}
          <img src={inspectImage} className="my-2" style={{ width: "5%" }} alt="Magnifying Glass" />
          <a href ="https://drive.google.com/file/d/1Mq01MP39wKnPAV0HNqoXiqBHhvkNsTw2/view?usp=drive_link"  
                    target="_blank" rel="noopener noreferrer" aria-label="My-Resume" id="my-resume">
                      Click here to inspect my complete Resume. </a>
          <img src={inspectImage} className="my-2" style={{ width: "5%" }} alt="Magnifying Glass" />
        </div>

        {/* Define the overview/summary section, just above the lists that will be animated. */}
        <div className="my-2 justify-content-center">
          <p>
            I am an Engineer focused on (Engineering) software development.  I have developed software in a
            variety of languages, presented usage scenarios in seminars, developed the associated documentation
            and marketing materials, and supported users with application issues.
          </p>
        </div>

        {/* Define two sets of 'bullets', describing the Engineering and Software capabilities. */}
        <ReactCardFlip isFlipped={isFlipped} flipDirection="vertical">
            <div className="front location-front-item flippers">
              <h5 className="my-5 ">&nbsp; Software Expertise:</h5>
              <ul>
                <li id="skill-list">Languages (C, C++, Fortran, Assembly)</li>
                <li id="skill-list">Front-end web technologies (HTML, CSS, Python, Javascript, React)</li>
                <li id="skill-list">Development tools (Git, npm, Jest, Webpack)</li>
                <li id="skill-list">Database technologies (SQLite, MySQL, MongoDB)</li>
                <li id="skill-list">Back-end technologies (Express, NodeJS)</li>
              </ul>
              <button className="front-button" onClick={handleFlip}>
                  <FcEngineering size={30} color="#ff9800"></FcEngineering>&nbsp; Engineering
              </button>
            </div>

            <div className="back location-back-item flippers">
              <h5 className="my-5">&nbsp; Engineering Expertise:</h5>
              <ul>
                <li id="skill-list">Education (BS Ocean Engineering, MS Civil Engineering, MBA)</li>
                <li id="skill-list">Certificates (Online Coding, UT Austin; Financial Management, Johns Hopkins Baltimore)</li>
                <li id="skill-list">Licensing: Texas Professional Engineer</li>
                <li id="skill-list">Society Membership (ASCE, ASME)</li>
                <li id="skill-list">Skilled Presenter (Webinars and Seminars)</li>
                <li id="skill-list">Competencies (Piping codes, Equipment codes, Seismic and Wind standards)</li>
              </ul>
  
              <button className="back-button" onClick={handleFlip}>
                  <HiOutlineDesktopComputer size={30} color="#ff9800"></HiOutlineDesktopComputer>&nbsp; Software
              </button>
            </div>
        </ReactCardFlip>
        
      </div>
    </section>
  )
}

export default Expertise