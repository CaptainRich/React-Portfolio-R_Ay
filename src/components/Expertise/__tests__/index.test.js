// __tests__/About.test.js
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Expertise from '..';

// Invoke the cleanup function, to remove components from the DOM to prevent
// memory leaks as well as variable/data collisions between tests.
afterEach( cleanup );

// Declare the component we're testing
describe( "Expertise component", () =>{
    
    // First test - baseline to verify the component is rendering
    it('renders', () => {
        render(<Expertise />);
      });

    // Second test - compare a snapshot of the 'About' component
    it('matches snapshot DOM node structure', () => {
        
        const { asFragment } = render(<Expertise />);  // return a snapshot of the 'About' component
        expect(asFragment()).toMatchSnapshot();    // do actual and expected snapshots math
      });

});