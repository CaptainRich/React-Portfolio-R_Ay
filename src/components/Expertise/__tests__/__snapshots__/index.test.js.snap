// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expertise component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section
    class="my-5"
  >
    <h1
      id="expertise"
    >
      Capabilities & Skills ...
    </h1>
    <h1>
      <p />
       
    </h1>
    <div
      class="container"
    >
      <div
        class="my-resume"
      >
        <img
          alt="Magnifying Glass"
          class="my-2"
          src="mag-glass.svg"
          style="width: 5%;"
        />
        <a
          aria-label="My-Resume"
          href="https://drive.google.com/file/d/1ckQtj7fc67mH15Ucy8NuR81DeRe0bwQ3/view?usp=sharing"
          id="my-resume"
          rel="noopener noreferrer"
          target="_blank"
        >
          Click here to inspect my complete Resume. 
        </a>
        <img
          alt="Magnifying Glass"
          class="my-2"
          src="mag-glass.svg"
          style="width: 5%;"
        />
      </div>
      <div
        class="my-2 justify-content-center"
      >
        <p>
          I am an Engineer focused on (Engineering) software development.  I have developed software in a variety of languages, presented usage scenarios in seminars, developed the associated documentation and marketing materials, and supported users with application issues.
        </p>
      </div>
      <div
        class="react-card-flip"
        style="z-index: auto;"
      >
        <div
          class="react-card-flipper"
          style="height: 100%; position: relative; width: 100%;"
        >
          <div
            class="react-card-front"
            style="height: 100%; left: 0px; position: relative; top: 0px; transform: rotateX(0deg); transition: 0.6s; width: 100%; z-index: 2;"
          >
            <div
              class="front location-front-item flippers"
            >
              <h5
                class="my-5 "
              >
                  Software Expertise:
              </h5>
              <ul>
                <li
                  id="skill-list"
                >
                  Languages (C, C++, Fortran, Assembly)
                </li>
                <li
                  id="skill-list"
                >
                  Front-end web technologies (HTML, CSS, Javascript, React)
                </li>
                <li
                  id="skill-list"
                >
                  Development tools (Git, npm, Jest, Webpack)
                </li>
                <li
                  id="skill-list"
                >
                  Database technologies (SQLite, MySQL, MongoDB)
                </li>
                <li
                  id="skill-list"
                >
                  Back-end technologies (Express, NodeJS)
                </li>
              </ul>
              <button
                class="front-button"
              >
                <svg
                  color="#ff9800"
                  enable-background="new 0 0 48 48"
                  fill="currentColor"
                  height="30"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: rgb(255, 152, 0);"
                  version="1"
                  viewBox="0 0 48 48"
                  width="30"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M37.4,24.6l-11.6-2.2l-3.9-11.2l-3.8,1.3L22,23.6l-7.8,9l3,2.6l7.8-9l11.6,2.2L37.4,24.6z"
                    fill="#EF6C00"
                  />
                  <g
                    fill="#FF9800"
                  >
                    <path
                      d="M24,19c-2.8,0-5,2.2-5,5c0,2.8,2.2,5,5,5s5-2.2,5-5C29,21.2,26.8,19,24,19z M24,26c-1.1,0-2-0.9-2-2 c0-1.1,0.9-2,2-2s2,0.9,2,2C26,25.1,25.1,26,24,26z"
                    />
                    <path
                      d="M40.7,27c0.2-1,0.3-2,0.3-3c0-1-0.1-2-0.3-3l3.3-2.4c0.4-0.3,0.6-0.9,0.3-1.4L40,9.8 c-0.3-0.5-0.8-0.7-1.3-0.4L35,11c-1.5-1.3-3.3-2.3-5.2-3l-0.4-4.1c-0.1-0.5-0.5-0.9-1-0.9h-8.6c-0.5,0-1,0.4-1,0.9L18.2,8 c-1.9,0.7-3.7,1.7-5.2,3L9.3,9.3C8.8,9.1,8.2,9.3,8,9.8l-4.3,7.4c-0.3,0.5-0.1,1.1,0.3,1.4L7.3,21C7.1,22,7,23,7,24 c0,1,0.1,2,0.3,3L4,29.4c-0.4,0.3-0.6,0.9-0.3,1.4L8,38.2c0.3,0.5,0.8,0.7,1.3,0.4L13,37c1.5,1.3,3.3,2.3,5.2,3l0.4,4.1 c0.1,0.5,0.5,0.9,1,0.9h8.6c0.5,0,1-0.4,1-0.9l0.4-4.1c1.9-0.7,3.7-1.7,5.2-3l3.7,1.7c0.5,0.2,1.1,0,1.3-0.4l4.3-7.4 c0.3-0.5,0.1-1.1-0.3-1.4L40.7,27z M24,35c-6.1,0-11-4.9-11-11c0-6.1,4.9-11,11-11s11,4.9,11,11C35,30.1,30.1,35,24,35z"
                    />
                  </g>
                </svg>
                  Engineering
              </button>
            </div>
          </div>
          <div
            class="react-card-back"
            style="height: 100%; left: 0px; position: absolute; top: 0px; transform: rotateX(-180deg); transition: 0.6s; width: 100%;"
          >
            <div
              class="back location-back-item flippers"
            >
              <h5
                class="my-5"
              >
                  Engineering Expertise:
              </h5>
              <ul>
                <li
                  id="skill-list"
                >
                  Education (BS Ocean Engineering, MS Civil Engineering, MBA)
                </li>
                <li
                  id="skill-list"
                >
                  Licensing: Texas Professional Engineer
                </li>
                <li
                  id="skill-list"
                >
                  Society Membership (ASCE, ASME)
                </li>
                <li
                  id="skill-list"
                >
                  Skilled Presenter (Webinars and Seminars)
                </li>
                <li
                  id="skill-list"
                >
                  Competencies (Piping codes, Seismic and Wind standards)
                </li>
              </ul>
              <button
                class="back-button"
              >
                <svg
                  color="#ff9800"
                  fill="none"
                  height="30"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: rgb(255, 152, 0);"
                  viewBox="0 0 24 24"
                  width="30"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.75 17L9 20l-1 1h8l-1-1-.75-3M3 13h18M5 17h14a2 2 0 002-2V5a2 2 0 00-2-2H5a2 2 0 00-2 2v10a2 2 0 002 2z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                  Software
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
